src/acachecontrol/cache.py

```diff
@@ -5,20 +5,39 @@
 import asyncio
 import hashlib
 import json
+import logging
 import time
 from typing import Any, Dict, Tuple
 
 from .exceptions import CacheException, TimeoutException
 
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_MAX_AGE = 120  # seconds
+
 
 class AsyncCache:
-    def __init__(self):
+    def __init__(self, config={}):
         self.cache = {}  # type: Dict
         self._wait_until_completed = set()
+        self.default_max_age = config.get("max-age", DEFAULT_MAX_AGE)
 
     def __contains__(self, key):
         return self._make_key_hashable(key) in self.cache
 
+    def has_valid_entry(self, key):
+        """Check if entry exists and not expired, delete expired."""
+        if key in self:
+            cache_key = self._make_key_hashable(key)
+            if self.is_cache_entry_expired(self.cache[cache_key]):
```

Comment:
```suggestion
            if self.is_cache_entry_expired(key):
```

Move `cache_key = self._make_key_hashable(key)` to `is_cache_entry_expired()`. WDYT?

Reply:
agree, sounds logically